apps/docs/docs/reference/javascript/protocol.mdx

```diff
@@ -0,0 +1,152 @@
+---
+id: protocol
+title: Realtime Protocol
+description: Understanding Realtime Protocol
+sidebar_label: Protocol
+---
+
+The Realtime Protocal is a set of message formats that you send and receive from the Realtime server.
+
+## Connection
+
+In the initial message, the client sends a message specifying the features that wants to use (Broadcase, Presence, Postgres Changes).
+
+```typescript
+{
+   "event": "phx_join",
+   "topic": string,
+   "payload": {
+      "config": {
+         "broadcast": {
+            "self": boolean
+         },
+         "presence": {
+            "key": string
+         },
+         "postgres_changes": [
+            {
+               "event": "*" | "INSERT" | "UPDATE" | "DELETE",
+               "schema": string,
+               "table": string,
+               "filter": string + '=' + "eq" | "neq" | "gt" | "gte" | "lt" | "lte" +  '.' + string
+            }
+         ]
+      }
+   },
+   "ref": string
+}
+```
+
+In response, the server sends the Postgres configuration with a unique ID. With this ID, the client should route incoming changes to the appropriate callback.
+
+```typescript
+{
+   "event": "phx_reply",
+   "topic": string,
+   "payload": {
+      "response": {
+         "postgres_changes": [
+            {
+               "id": number,
+               "event": "*" | "INSERT" | "UPDATE" | "DELETE",
+               "schema": string,
+               "table": string,
+               "filter": string + '=' + "eq" | "neq" | "gt" | "gte" | "lt" | "lte" +  '.' + string
```

Comment:
@dannykng can we get your feedback on this type?

Reply:
Replied in Slack (to consolidate the Realtime docs workstreams).